internal/httputil/pagination.go

```diff
@@ -0,0 +1,58 @@
+package httputil
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+)
+
+const maxPageSize = 100
+
+type PaginationData struct {
+	PageNumber int
+	PageSize   int
+	SortBy     map[string]string
+	FilterBy   map[string]string
+}
+
+func ParsePaginationQuery(r *http.Request) (*PaginationData, error) {
+	qs := r.URL.Query()
+
+	pagination := &PaginationData{
+		PageNumber: 0,
+		PageSize:   10,
+	}
+
+	pageNumber := qs.Get("page")
+	if pageNumber != "" {
```

Comment:
This if clause is redundant. You get an error from strconv.Atoi if you pass empty string into it.